app/src/main/kotlin/org/fossify/clock/dialogs/ExportDataDialog.kt

```diff
@@ -0,0 +1,72 @@
+package org.fossify.clock.dialogs
+
+import androidx.appcompat.app.AlertDialog
+import org.fossify.commons.activities.BaseSimpleActivity
+import org.fossify.commons.dialogs.FilePickerDialog
+import org.fossify.commons.extensions.*
+import org.fossify.commons.helpers.ensureBackgroundThread
+import org.fossify.clock.R
+import org.fossify.clock.databinding.DialogExportDataBinding
+import org.fossify.clock.extensions.config
+import org.fossify.clock.helpers.DATA_EXPORT_EXTENSION
+import java.io.File
+
+class ExportDataDialog(
+    val activity: BaseSimpleActivity,
+    val path: String,
+    val hidePath: Boolean,
+    callback: (file: File) -> Unit,
+) {
+    private var realPath = path.ifEmpty { activity.internalStoragePath }
+    private val config = activity.config
+
+    init {
+        val view = DialogExportDataBinding.inflate(activity.layoutInflater, null, false).apply {
+            exportDataFolder.text = activity.humanizePath(realPath)
+            exportDataFilename.setText("${activity.getString(R.string.settings_export_data)}_${activity.getCurrentFormattedDateTime()}")
```

Comment:
The filename turns out to be `Export alarms and timers...`. Let's hardcode the filename as done in other apps. 

```suggestion
            exportDataFilename.setText("alarms_and_timers_${activity.getCurrentFormattedDateTime()}")
```